task07_3/task07_3/Program.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace task07_3
+{
+    internal class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Введите позицию белой ладьи (например, a1):");
+            var whiteRookPosition = Console.ReadLine();
+
+            if (!IsPositionCorrect(whiteRookPosition))
+            {
+                Console.WriteLine("Некорректная позиция ладьи");
+                Console.ReadKey();
+                return;
+            }
+
+            Console.WriteLine("Введите позицию черного коня (например, a1):");
+            var blackKnightPosition = Console.ReadLine();
+
+            if (!IsPositionCorrect(blackKnightPosition) || whiteRookPosition == blackKnightPosition)
+            {
+                Console.WriteLine("Некорректная позиция черного коня");
+                Console.ReadKey();
+                return;
+            }
+
+            if (IsKnightStrike(blackKnightPosition, whiteRookPosition))
+            {
+                Console.WriteLine("Черный конь атакует белую ладью.");
+            }
+            else
+            {
+                Console.WriteLine("Черный конь не атакует белую ладью.");
+                Console.WriteLine("Введите ход белой ладьи (например, a3):");
+                var move = Console.ReadLine();
+
+                if (IsRookMoveValid(whiteRookPosition, move))
+                    Console.WriteLine("Ход разрешен");
```

Comment:
Нет проверки на то, сто ладья попадет под бой коня
```suggestion
                if (IsRookMoveValid(whiteRookPosition, move)  && 
                        !IsKnightStrike(blackKnightPosition,move))
                    Console.WriteLine("Ход разрешен");
```